newswires/app/db/FingerpostWireEntry.scala

```diff
@@ -151,24 +172,35 @@ object FingerpostWireEntry extends SQLSyntaxSupport[FingerpostWireEntry] {
       )
     ).flatten
 
+    val highlightsClause = maybeFreeTextQuery match {
+      case Some(query) =>
+        sqls", ts_headline('simple', $contentCol->>'body_text', phraseto_tsquery($query)) AS ${syn.result.highlight}"
+      case None => sqls""
```

Comment:
this seems to be really awkward; the fixes might be to run a completely different query when we expect a highlight back (and extracting all the common query bits into something shared), but we can also go hack with
```suggestion
      case None => sqls"'' AS ${syn.resultName.highlight}"
```
(ie. always return an empty string as a "highlight", even if there is no query to highlight)

Reply:
ooh, interesting! thanks for looking at this!